platform/ios/src/MGLMapView.mm

```diff
@@ -4729,7 +4736,7 @@ - (void)notifyMapChange:(mbgl::MapChange)change
                             rect = annotationView.frame;
                         }
 
-                        if ( ! CGRectIntersectsRect(rect, self.frame)) {
+                        if ( ! CGRectIntersectsRect(rect, self.frame) && self.deselectWhenOutOfBounds) {
```

Comment:
Cool, I have updated my PR to that annotations are not deselected any longer. If the annotation has a callout view which responds to `dismissAutomatically`, the annotation is deselected if that function returns `true`.